### ISSUE TITLE
In tutorial_train_mnist.py, what is the reason for choosing 'mean' or 'sum' in F.cross_entropy

### ISSUE THREAD

**Issue by TianQi-777** [BorealisAI/advertorch#56]
In 'tutorial_train_mnist.py', there are 4 `F.cross_entrop`, they are located in 4 places below.
[place 1(Line 60)](https://github.com/BorealisAI/advertorch/blob/5daeeee98077afacf94728936a04cebde474259f/advertorch_examples/tutorial_train_mnist.py#L60)
```
    if flag_advtrain:
        from advertorch.attacks import LinfPGDAttack
        adversary = LinfPGDAttack(
            model, loss_fn=nn.CrossEntropyLoss(reduction="sum"), eps=0.3,     # Use "sum"
            nb_iter=40, eps_iter=0.01, rand_init=True, clip_min=0.0,
            clip_max=1.0, targeted=False)
```
[place 2(Line 78)](https://github.com/BorealisAI/advertorch/blob/5daeeee98077afacf94728936a04cebde474259f/advertorch_examples/tutorial_train_mnist.py#L78)
```
            optimizer.zero_grad()
            output = model(data)
            loss = F.cross_entropy(
                output, target, reduction='elementwise_mean')                                       # Use "mean"
            loss.backward()
            optimizer.step()
```
[place 3(Line 100)](https://github.com/BorealisAI/advertorch/blob/5daeeee98077afacf94728936a04cebde474259f/advertorch_examples/tutorial_train_mnist.py#L100)
```
        for clndata, target in test_loader:
            clndata, target = clndata.to(device), target.to(device)
            with torch.no_grad():
                output = model(clndata)
            test_clnloss += F.cross_entropy(
                output, target, reduction='sum').item()                                                           # Use "sum"
            pred = output.max(1, keepdim=True)[1]
            clncorrect += pred.eq(target.view_as(pred)).sum().item()
```
[place 4(Line 109)](https://github.com/BorealisAI/advertorch/blob/5daeeee98077afacf94728936a04cebde474259f/advertorch_examples/tutorial_train_mnist.py#L109)
```
            if flag_advtrain:
                advdata = adversary.perturb(clndata, target)
                with torch.no_grad():
                    output = model(advdata)
                test_advloss += F.cross_entropy(
                    output, target, reduction='sum').item()                                                      # Use "sum"
                pred = output.max(1, keepdim=True)[1]
                advcorrect += pred.eq(target.view_as(pred)).sum().item()
```
Can using "mean" in `place 1` affect the results? Or why use "mean"(`place 2`) when updating the model and "sum"(`place 1`) when generating noise

**Comment by gwding** [comment]
The gist is the when you're training the model, you are using the gradient on the parameter, and you don't want batch size to affect "average" magnitude of the gradient, and using `mean` can achieve that.

Similarly, when attacking, you are using the gradient on the input, and you don't want batch size to affect magnitude of the gradient on each input, and using `sum` can achieve that.

The 3rd and 4th are for book keeping purposes, the sum got simply averaged at then end.

**Comment by TianQi-777** [comment]
Thank you for your reply, I understand it now.
